state/protocol/badger/snapshot.go

```diff
@@ -274,6 +278,18 @@ func (s *Snapshot) SealingSegment() (*flow.SealingSegment, error) {
 		return nil, protocol.ErrSealingSegmentBelowRootBlock
 	}
 
+	// Verify that head of sealing segment is finalized.
+	finalizedBlockAtHeight, err := s.state.headers.BlockIDByHeight(head.Header.Height)
```

Comment:
can we add test cases to cover those paths where we return sentinel errors?

Reply:
you are totally right. Working on the tests ...

Reply:
implemented in https://github.com/onflow/flow-go/pull/3824/commits/47b7cab013e9fde0b717a31946b9a58b2a8408f3